File: src/main/java/hacker/solution/ExceptionDemo.java
package hacker.solution;

import java.util.InputMismatchException;
import java.util.Scanner;

public class ExceptionDemo {
    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        try {
            int x = scan.nextInt();
            int y = scan.nextInt();
            System.out.println(x/y);
        } catch (InputMismatchException e) {
            System.out.println("java.util.InputMismatchException");
        } catch (ArithmeticException e) {
            System.out.println("java.lang.ArithmeticException: "+e.getMessage());
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}

File: src/main/kotlin/hacker/solutionk/ExceptionDemo.kt
package hacker.solutionk

import java.util.*

object ExceptionDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`);
        try {
            val x = sc.nextInt();
            val y = sc.nextInt();
            println(x/y);
        } catch (e: InputMismatchException) {
            println("java.util.InputMismatchException");
        } catch (e: ArithmeticException) {
            println("java.lang.ArithmeticException: " + e.message)
        } catch (e: Exception) {
            println(e.message)
        }
    }
}